In [2]:
import pandas as pd 
import numpy as np
import matplotlib as plt
import seaborn as sas
import os

In [2]:
visa_df = pd.read_csv('Visadataset.csv')
visa_df

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
4                         N             1082         2005   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
4                    South      149907.3900         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
4       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[25480 rows x 12 columns]

#### Divide into Categorical and Numerical columns

In [4]:
visa_df.dtypes

case_id                   object
continent                 object
education_of_employee     object
has_job_experience        object
requires_job_training     object
no_of_employees            int64
yr_of_estab                int64
region_of_employment      object
prevailing_wage          float64
unit_of_wage              object
full_time_position        object
case_status               object
dtype: object

In [5]:
types = dict(visa_df.dtypes)
categorical_visa_df = [key for key,value in types.items() if value == 'object']
numerical_visa_df = [key for key,value in types.items() if value != 'object']
print(categorical_visa_df)

['case_id', 'continent', 'education_of_employee', 'has_job_experience', 'requires_job_training', 'region_of_employment', 'unit_of_wage', 'full_time_position', 'case_status']


In [6]:
print (numerical_visa_df)

['no_of_employees', 'yr_of_estab', 'prevailing_wage']


In [7]:
categorical_visa_df_sel = visa_df.select_dtypes(include='object').columns
categorical_visa_df_sel

Index(['case_id', 'continent', 'education_of_employee', 'has_job_experience',
       'requires_job_training', 'region_of_employment', 'unit_of_wage',
       'full_time_position', 'case_status'],
      dtype='object')

In [8]:
numerical_visa_df_sel = visa_df.select_dtypes(exclude='object').columns
numerical_visa_df_sel

Index(['no_of_employees', 'yr_of_estab', 'prevailing_wage'], dtype='object')

In [9]:
categorical_visa_df_sel_values = visa_df.select_dtypes(include='object')
categorical_visa_df_sel_values

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
4         EZYV05    Africa              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training region_of_employment unit_of_wage  \
0                         N                 West         Hour   
1                         N            Northeast         Year   
2                         Y                 West         Year   
3                         N                 West         Year   
4                         N                South         Year   
...                     ...                  ...          ...   
25475                     Y                South         Year   
25476                     N            Northeast         Year   
25477                     N                South         Year   
25478                     Y                 West         Year   
25479                     N              Midwest         Year   

      full_time_position case_status  
0                      Y      Denied  
1                      Y   Certified  
2                      Y      Denied  
3                      Y      Denied  
4                      Y   Certified  
...                  ...         ...  
25475                  Y   Certified  
25476                  Y   Certified  
25477                  N   Certified  
25478                  Y   Certified  
25479                  Y   Certified  

[25480 rows x 9 columns]

In [10]:
float_visa_df = visa_df.select_dtypes(include='float64')
float_visa_df

prevailing_wage
0             592.2029
1           83425.6500
2          122996.8600
3           83434.0300
4          149907.3900
...                ...
25475       77092.5700
25476      279174.7900
25477      146298.8500
25478       86154.7700
25479       70876.9100

[25480 rows x 1 columns]

$Task-1$
- **we want to find the number of applicants continent wise**
- How many number of application from Asia out of 25480

$Unique$
- Unique will provide how many unique items are there in a particular column

In [13]:
visa_df['continent']

0          Asia
1          Asia
2          Asia
3          Asia
4        Africa
          ...  
25475      Asia
25476      Asia
25477      Asia
25478      Asia
25479      Asia
Name: continent, Length: 25480, dtype: object

In [14]:
visa_df_unique = visa_df['continent'].unique()
visa_df_unique

array(['Asia', 'Africa', 'North America', 'Europe', 'South America',
       'Oceania'], dtype=object)

In [15]:
len(visa_df['continent'].unique())

6

In [16]:
visa_df_count = visa_df['continent'].nunique()
visa_df_count

6

**Nnique**
- gives count of unique items

1. Read the data with specific column
2. Apply the condition
    - when we apply the condition python will give True or False
3. In order to retrive the True data
    - dataframe_name[con]
4. visa_df['continent']
5. con = visa_df['continent'] == 'Asia'
6. len(visa_df[con])

In [19]:
visa_df_asia_df = visa_df['continent'] == 'Asia'
visa_df_asia_df

0         True
1         True
2         True
3         True
4        False
         ...  
25475     True
25476     True
25477     True
25478     True
25479     True
Name: continent, Length: 25480, dtype: bool

- if we want to retrieve data from the the above follow this

In [21]:
len(visa_df[visa_df_asia_df])

16861

In [22]:
visa_df[visa_df_asia_df]

case_id continent education_of_employee has_job_experience  \
0         EZYV01      Asia           High School                  N   
1         EZYV02      Asia              Master's                  Y   
2         EZYV03      Asia            Bachelor's                  N   
3         EZYV04      Asia            Bachelor's                  N   
5         EZYV06      Asia              Master's                  Y   
...          ...       ...                   ...                ...   
25475  EZYV25476      Asia            Bachelor's                  Y   
25476  EZYV25477      Asia           High School                  Y   
25477  EZYV25478      Asia              Master's                  Y   
25478  EZYV25479      Asia              Master's                  Y   
25479  EZYV25480      Asia            Bachelor's                  Y   

      requires_job_training  no_of_employees  yr_of_estab  \
0                         N            14513         2007   
1                         N             2412         2002   
2                         Y            44444         2008   
3                         N               98         1897   
5                         N             2339         2012   
...                     ...              ...          ...   
25475                     Y             2601         2008   
25476                     N             3274         2006   
25477                     N             1121         1910   
25478                     Y             1918         1887   
25479                     N             3195         1960   

      region_of_employment  prevailing_wage unit_of_wage full_time_position  \
0                     West         592.2029         Hour                  Y   
1                Northeast       83425.6500         Year                  Y   
2                     West      122996.8600         Year                  Y   
3                     West       83434.0300         Year                  Y   
5                    South       78252.1400         Year                  Y   
...                    ...              ...          ...                ...   
25475                South       77092.5700         Year                  Y   
25476            Northeast      279174.7900         Year                  Y   
25477                South      146298.8500         Year                  N   
25478                 West       86154.7700         Year                  Y   
25479              Midwest       70876.9100         Year                  Y   

      case_status  
0          Denied  
1       Certified  
2          Denied  
3          Denied  
5       Certified  
...           ...  
25475   Certified  
25476   Certified  
25477   Certified  
25478   Certified  
25479   Certified  

[16861 rows x 12 columns]

In [23]:
for i in visa_df_unique:
    print (i)

Asia
Africa
North America
Europe
South America
Oceania


In [24]:
for i in visa_df_unique:
    con = visa_df['continent'] == i
    count = len(visa_df[con])
    print (f'The number of applications from {i} is: {count}')

The number of applications from Asia is: 16861
The number of applications from Africa is: 551
The number of applications from North America is: 3292
The number of applications from Europe is: 3732
The number of applications from South America is: 852
The number of applications from Oceania is: 192


In [50]:
visa_df_unique_continents = visa_df['continent'].unique()
count = []
for i in visa_df_unique_continents:
    con = visa_df['continent'] == i
    count.append (len (visa_df [con]))


cols = ['Continents','Number of applications']
continents_df = pd.DataFrame(zip(visa_df_unique,count),columns=cols)

In [52]:
continents_df

Continents  Number of applications
0           Asia                   16861
1         Africa                     551
2  North America                    3292
3         Europe                    3732
4  South America                     852
5        Oceania                     192

In [56]:
continents_df.to_csv('Continents_df.csv',index=False)